packages/react/src/products/hooks/useProductGrouping.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  buildQueryStringFromObject,
+  fetchProductGrouping,
+  getProductGrouping,
+  getProductGroupingError,
+  isProductGroupingFetched,
+  isProductGroupingLoading,
+  ProductEntity,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { UseProductGroupingOptions } from './types';
+
+const useProductGrouping = (
+  productId: ProductEntity['id'],
+  options: UseProductGroupingOptions = {},
+) => {
+  const {
+    fetchConfig,
+    enableAutoFetch = true,
+    fetchQuery = { pageIndex: 1 },
+  } = options;
+  const fetchAction = useAction(fetchProductGrouping);
+
+  const error = useSelector((state: StoreState) =>
+    getProductGroupingError(state, productId),
```

Comment:
Shouldn't these selectors receive the query hash as well? If the fetchQuery changes between renders (or different components that are rendered at the same time use different query parameters), it will not reflect the state of the correct request. This is valid for `isProduuctGroupingLoading` and `isProductGroupingFetched` selectors as well.